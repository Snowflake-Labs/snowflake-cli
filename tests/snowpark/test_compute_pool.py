from unittest import mock


@mock.patch("snowflake.connector.connect")
def test_create_cp(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(
        [
            "snowpark",
            "cp",
            "create",
            "--name",
            "cpName",
            "--num",
            "42",
            "--family",
            "familyValue",
        ]
    )

    assert result.exit_code == 0
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_list_cp(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "cp", "list"])

    assert result.exit_code == 0
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_drop_cp(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "cp", "drop", "cpNameToDrop"])

    assert result.exit_code == 0
    assert ctx.get_query() == snapshot


@mock.patch("snowflake.connector.connect")
def test_stop_cp(mock_connector, runner, mock_ctx, snapshot):
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    result = runner.invoke(["snowpark", "cp", "stop", "cpNameToStop"])

    assert result.exit_code == 0
    assert ctx.get_query() == snapshot
