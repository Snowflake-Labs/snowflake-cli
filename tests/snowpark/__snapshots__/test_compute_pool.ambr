# serializer version: 1
# name: test_create_cp
  '''
  CREATE COMPUTE POOL cpName
  MIN_NODES = 42
  MAX_NODES = 42
  INSTANCE_FAMILY = familyValue;
  
  '''
# ---
# name: test_drop_cp
  'drop compute pool cpNameToDrop;'
# ---
# name: test_list_cp
  'show compute pools;'
# ---
# name: test_stop_cp
  'alter compute pool cpNameToStop stop all services;'
# ---
